packages/components/src/internal/components/editable/EditableGridPanelForUpdateWithLineage.tsx

```diff
@@ -0,0 +1,337 @@
+            dataModels: QueryModel[];
+            editorModels: EditorModel[];
+        }> => {
+            return await initEditableGridModels(editableGridModels.dataModels, editableGridModels.editorModels, queryModel, loaders);
+        };
+
+        if (loaders && editableGridModels?.dataModels?.find(dataModel => dataModel.isLoading)) {
+            initEditorModel().then(models => {
+                setEditableGridModels(models)
+            }).catch(error => {
+                createNotification({
+                    message: error,
+                    alertClass: "danger"
+                })
+            })
+        }
+    }, [loaders, editableGridModels]);
+
+    const onGridChange = useCallback(
+        (
+            editorModelChanges: Partial<EditorModelProps>,
+            dataKeys?: List<any>,
+            data?: Map<string, Map<string, any>>,
+            index?: number
+        ): void => {
+            setEditableGridModels(_models =>
+                applyEditableGridChangesToModels(
+                    _models.dataModels,
+                    _models.editorModels,
+                    editorModelChanges,
+                    undefined,
+                    dataKeys,
+                    data,
+                    index ?? 0
+                )
+            );
+        },
+        []
+    );
+
+    const onSubmit = useCallback((): void => {
+        const gridDataAllTabs = [];
+        editableGridModels.dataModels.forEach((model, ind) => {
+            const gridData = getUpdatedDataFromEditableGrid(
+                editableGridModels.dataModels,
+                editableGridModels.editorModels,
+                idField,
+                readOnlyColumns,
+                selectionData,
+                ind
+            );
+            if (gridData) {
+                gridDataAllTabs.push(gridData);
+            }
+        });
+
+        if (gridDataAllTabs.length > 0) {
+            setIsSubmitting(true);
+            updateAllTabRows(gridDataAllTabs).then(result => {
+                setIsSubmitting(false);
+                if (result !== false) {
+                    onComplete();
+                }
+            });
+        } else {
+            setIsSubmitting(false);
+            onComplete();
+        }
+    }, [editableGridModels, onComplete]);
+
+    const getCurrentTab = useCallback(
+        (tabInd: number): number => {
+            return tabInd === undefined ? includedTabs[0] : includedTabs[tabInd];
+        },
+        [includedTabs]
+    );
+
+    const removeParentType = useCallback(
+        (index: number, queryName: string): void => {
+            const tabIndex = includedTabs.indexOf(UpdateGridTab.Lineage);
+
+            const { editorModelChanges, data, queryInfo, entityParents } = removeEntityParentType(
+                index,
+                queryName,
+                entityParentsMap,
+                editableGridModels.editorModels[tabIndex],
```

Comment:
Upon successfully deleting I feel like we should display a notification. I believe this would be consistent with Sample Types, etc.

Reply:
So this function is just for removing a parent type drop down selection from the grid header in the lineage grid.  It does not get called when actually saving data.  There is a "Successfully updated..." message when lineage is updated/deleted in the grid and saved.  Is there somewhere in the UI you're not seeing a message?

Reply:
Heh, I'm seeing a notification now. Unsure if this changed but I likely just missed it.